src/main/java/org/aerogear/kryptowire/workflow/KWSubmitStep.java

```diff
@@ -0,0 +1,111 @@
+import org.aerogear.kryptowire.GlobalConfigurationImpl;
+import org.aerogear.kryptowire.KryptowireService;
+import org.aerogear.kryptowire.KryptowireServiceImpl;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+public class KWSubmitStep extends AbstractStepImpl {
+
+    private String platform;
+    private String filePath;
+
+    public String getPlatform() {
+        return platform;
+    }
+
+    @DataBoundSetter
+    public void setPlatform(String platform) {
+        this.platform = platform;
+    }
+
+    public String getFilePath() {
+        return filePath;
+    }
+
+    @DataBoundSetter
+    public void setFilePath(String filePath) {
+        this.filePath = filePath;
+    }
+
+    @DataBoundConstructor
+    public KWSubmitStep(@Nonnull String platform, @Nonnull String filePath) {
+        this.platform = platform;
+        this.filePath = filePath;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(KWSubmitExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "kwSubmit";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Submit to Kryptowire";
+        }
+
+    }
+
+    public static class KWSubmitExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        private static final long serialVersionUID = 1L;
+
+        @Inject
+        transient KWSubmitStep step;
+
+        @StepContextParameter
+        transient TaskListener listener;
+
+        @Override
+        protected Void run() throws Exception {
+
+            GlobalConfigurationImpl pluginConfig = GlobalConfiguration.all().get(GlobalConfigurationImpl.class);
+
+            String kwEndpoint = pluginConfig.getKwEndpoint();
+            String kwApiKey = pluginConfig.getKwApiKey();
+
+            if(StringUtils.isEmpty(kwEndpoint) || StringUtils.isEmpty(kwApiKey)) {
+                throw new RuntimeException("Kryptowire plugin configuration is not set!");
+            }
+
+            FilePath fp = getContext().get(FilePath.class).child(step.filePath);
+
+            listener.getLogger().println(" --- Kryptowire submit Start ---");
+            listener.getLogger().println("kwSubmit: " + step.platform + " : " + step.filePath);
+            KryptowireService kws = new KryptowireServiceImpl(kwEndpoint,  kwApiKey);
+            JSONObject resp = kws.submit(step.platform, fp);
```

Comment:
I think this the step we need to catch/look for an error (e.g. anything that is not a 2xx response status code) and throw a runtime error if something goes wrong

Reply:
The BasicResonseHandler used in the KryptowireServiceImpl submit method will throw a HttpResponseException and fail the build if anything other than a 2xx response is returned.

https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/client/BasicResponseHandler.html